src/wp-includes/option.php

```diff
@@ -965,7 +968,11 @@ function add_option( $option, $value = '', $deprecated = '', $autoload = 'yes' )
 	}
 
 	$serialized_value = maybe_serialize( $value );
-	$autoload         = ( 'no' === $autoload || false === $autoload ) ? 'no' : 'yes';
+
+	if ( null === $autoload ) {
+		$autoload = check_option_size( $serialized_value, $option );
+	}
+	$autoload = get_autoload_value( $autoload );
```

Comment:
See above, this function isn't necessary. Developers should not be able to pass `default-yes` or `default-no`, those values should be for internal handling only.
```suggestion
	} else {
		$autoload = ( 'no' === $autoload || false === $autoload ) ? 'no' : 'yes';
	}
```

Reply:
@felixarntz What does happen if a developer does pass those values? I know this is edge case, but it is is a valid a value, it does seem like a developer should be able to set it. There maybe some reason, like testing that would make doing this useful. Again, before removing this, I would like to see feedback from others.